src/Octoshift/Handlers/DownloadLogsCommandHandler.cs

```diff
@@ -75,20 +73,17 @@ public async Task Handle(DownloadLogsCommandArgs args)
             _log.LogWarning($"Overwriting {args.MigrationLogFile} due to --overwrite option.");
         }
 
-        var result = await _retryPolicy.RetryOnResult(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), string.Empty,
+        var result = await _retryPolicy.RetryOnResult<(string MigrationLogURL, string MigrationID)>(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), result => string.IsNullOrEmpty(result.MigrationLogURL),
             "Waiting for migration log to populate...");
 
-        if (result.Outcome == OutcomeType.Successful && result.Result is null)
-        {
-            throw new OctoshiftCliException($"Migration for repository {args.GithubRepo} not found!");
-        }
-
         if (result.Outcome == OutcomeType.Failure)
         {
             throw new OctoshiftCliException($"Migration log for repository {args.GithubRepo} unavailable!");
         }
 
-        var logUrl = result.Result;
+        (var logUrl, var migrationId) = result.Result;
+
+        args.MigrationLogFile ??= $"migration-log-{args.GithubOrg}-{args.GithubRepo}-{migrationId}.log";
```

Comment:
not sure if this is necessary because if we got to logging the log file, the `MigrationLogFile` will have a value otherwise we throw.

Reply:
The purpose of this line is to set the file name if the user didn't specify one when they called the command. Does that make sense?

Reply:
but how come it wasn't the case before?

Reply:
This line was already here, but I've moved it down in the file so that we can include the migration ID in the filename. Am I missing something?